Ardupilog.m

```diff
@@ -308,6 +314,53 @@ function delete(obj)
             return;
         end
 
+        function slice = getSlice(obj, slice_values, slice_type)
+        % This returns an indexed portion (a "slice") of an Ardupilog
+        % Example:
+        %    log_during_cruise = Log.getSlice([t_begin_cruise, t_end_cruise], 'TimeUS')
+        %  will return a smaller Ardupilog, only containing log data between
+        %  TimeUS values greater than t_begin_cruise and less than t_end_cruise.
+
+            % Copy all the properties, zero the number of messages
+            slice = copy(obj);
+            slice.numMsgs = 0;
+            
+            % Loop through the LogMsgGroups, slicing each one
+            for msgGroup = slice.logMsgGroups
```

Comment:
I'd prefer to avoid creating another global property when a local approach is possible.
But I don't want to make you jump through hoops.
I'll merge this now and add an issue for me to change this eventually with the 'isa()' approach you suggested.